### PR TITLE
feat: Add pagination to the groups table gf-83

### PR DESCRIPTION
**Pull request by Fjortis**
Implemented Pagination: Added pagination functionality to the group table.
Consistency with User Table: Adjusted the implementation to follow the pagination logic of the user table, ensuring uniformity across the application.
Shared Constants and Types: Utilized the same constants and types for pagination to maintain consistency.
Grouptablepag1 9

### CODE REVIEW

File: apps/backend/src/modules/groups/group.repository.ts [hunk 1]
```diff
@@ -45,13 +49,19 @@ class GroupRepository implements Repository {
...
+	public async findAll({
+		page,
+		pageSize,
+	}: PaginationQueryParameters): Promise<PaginationResponseDto<GroupEntity>> {
+		const { results, total } = await this.groupModel
 			.query()
+			.page(page, pageSize)
 			.withGraphFetched("[permissions, users]");
 
+		// Return the paginated response with mapped entities
```
Comment (Vitaliistf) [inline apps/backend/src/modules/groups/group.repository.ts:61]: I think that this comment can be removed, it is clear what we are returning from the signature.
```suggestion
```

File: apps/frontend/src/libs/components/table/styles.module.css [hunk 2]
```diff
@@ -40,7 +40,6 @@
 
 .table .table-body {
 	display: block;
-	max-height: 450px;
```
Comment (Vitaliistf) [inline apps/frontend/src/libs/components/table/styles.module.css:43]: You shouldn't change this. This is limited to make the table scrollable.

File: packages/shared/src/libs/types/pagination-response.type.ts
Comment (Vitaliistf) [inline packages/shared/src/libs/types/pagination-response.type.ts:1]: File name is not corresponding to the content, and we actually already have such type in `pagination-query-parameters.type.ts`. Check your exports and imports after deleting.

File: tsx
Comment (Vitaliistf) [inline tsx:1]: I think this file is also redundant.

File: apps/backend/src/modules/projects/project.service.ts [hunk 3]
```diff
@@ -68,7 +68,6 @@ class ProjectService implements Service {
 			items: projects.items.map((item) => item.toObject()),
 		};
 	}
-
 	public async findAllbyName(
```
Comment (kolibri753) [inline apps/backend/src/modules/projects/project.service.ts:71]: Better add that line back here, so this file won't be changed :sweat_smile:

File: packages/shared/src/libs/types/pagination-response.type.ts
Comment (Fjortis) [reply packages/shared/src/libs/types/pagination-response.type.ts:1]: Oh, that was an earlier implementation of my backend, i now cleaned it out, thanks for pointing it out :)

File: apps/frontend/src/libs/components/table/styles.module.css (hunk 2 above)
Comment (Fjortis) [reply apps/frontend/src/libs/components/table/styles.module.css:43]: You're right, will get this back to the orignal version

File: apps/backend/src/modules/groups/group.repository.ts (hunk 1 above)
Comment (Fjortis) [reply apps/backend/src/modules/groups/group.repository.ts]: Yep, it is now removed from the code

File: apps/backend/src/modules/projects/project.service.ts (hunk 3 above)
Comment (Fjortis) [reply apps/backend/src/modules/projects/project.service.ts:71]: The line has now been added back in :D now the PR is cleaner, thanks!

File: apps/frontend/src/libs/components/table-pagination/libs/constants/page-size-options.constant.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const PAGE_SIZE_OPTIONS = [
+	{ label: "5", value: 5 },
```
Comment (liza-veis) [inline apps/frontend/src/libs/components/table-pagination/libs/constants/page-size-options.constant.ts:2]: let's have 10 as minimum
```suggestion
```

File: apps/frontend/src/modules/groups/slices/group.slice.ts [hunk 5]
```diff
@@ -23,6 +25,7 @@ const { actions, name, reducer } = createSlice({
 		});
 		builder.addCase(loadAll.fulfilled, (state, action) => {
 			state.groups = action.payload.items;
+			state.totalGroupsCount = action.payload.totalItems;
```
Comment (liza-veis) [inline apps/frontend/src/modules/groups/slices/group.slice.ts:28]: the same as for users
```suggestion
			state.groupsTotalCount = action.payload.totalItems;
```

File: tsx
Comment (liza-veis) [inline tsx:1]: Remove this file

File: apps/frontend/src/pages/access-management/access-management.tsx [hunk 6]
```diff
@@ -62,17 +82,26 @@ const AccessManagement = (): JSX.Element => {
 				<div className={styles["users-table"]}>
 					<Table<UserRow> columns={userColumns} data={userData} />
 					<TablePagination
```
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/access-management.tsx:84]: What about query parameters for page and pageSize for both tables? they won't work properly with the same name, so we need to rename them

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/access-management/access-management.tsx (hunk 6 above)
Comment (Fjortis) [reply apps/frontend/src/pages/access-management/access-management.tsx:86]: This has now been adjusted, it now uses queryParameterPrefix to create the query parameters separetly for both tables
`	const {
		onPageChange: onGroupPageChange,
		onPageSizeChange: onGroupPageSizeChange,
		page: groupPage,
		pageSize: groupPageSize,
	} = usePagination({
		queryParameterPrefix: "group",
		totalItemsCount: groupsTotalCount,
	});`

File: tsx
Comment (Fjortis) [reply tsx:1]: Removed

File: apps/frontend/src/modules/groups/slices/group.slice.ts (hunk 5 above)
Comment (Fjortis) [reply apps/frontend/src/modules/groups/slices/group.slice.ts]: I've adjusted this to match users.

File: apps/frontend/src/libs/components/table-pagination/libs/constants/page-size-options.constant.ts (hunk 4 above)
Comment (Fjortis) [reply apps/frontend/src/libs/components/table-pagination/libs/constants/page-size-options.constant.ts]: I put it back to 10 as minimum

**Review by what1s1ove — APPROVED** [review]
lgtm after Lisa's comments

**Review by liza-veis — APPROVED** [review]

**Pull request merged**